drgn_tools/sysctl.py

```diff
@@ -18,11 +19,19 @@
 from drgn import Program
 from drgn.helpers.linux.rbtree import rbtree_inorder_for_each
 
+from drgn_tools.corelens import CorelensModule
 
-def get_sysctl_table(prog: Program) -> MappingProxyType:
+
+MAX_ELEMENTS = 100
+
+
+def get_sysctl_table(
+    prog: Program, limit: int = MAX_ELEMENTS
```

Comment:
Adding a `limit` parameter here breaks the caching. Suppose this function gets called with `limit=50`, and that result gets cached, and then later it is called with `limit=100`. The cached result is returned and it only has 50 entries.

Is there a use case for this yet? E.g. a particular sysctl that needs this? If not, I'd recommend removing the logic. But if there's a use case, let's take the `limit` and apply it in `print_sysctl_table()` rather than in `get_data_entry()`. So `get_data_entry()` would get the full value, and then the `print_sysctl_table()` would decide how much of it to print.

Reply:
Right. I don't think it's of much use. Removed it and made it just dump all.